apps/src/blockly/addons/cdoXml.js

```diff
@@ -148,26 +148,76 @@ export function addMutationToBehaviorDefBlocks(blockElement) {
   }
 }
 
+/**
+ * Adds a mutation element to a block if it's a text join block with an input count.
+ * CDO Blockly uses an unsupported method for serializing input count state
+ * where an arbitrary block attribute could be used to manage extra state.
+ * Mainline Blockly expects a mutator. The presence of the mutation element
+ * will trigger the block's domToMutation function to run, if it exists.
+ *
+ * @param {Element} blockElement - The XML element for a single block.
+ */
+export function addMutationToTextJoinBlocks(blockElement) {
+  if (
+    !['text_join', 'text_join_simple'].includes(
+      blockElement.getAttribute('type')
+    )
+  ) {
+    return;
+  }
+  const mutationElement =
+    blockElement.querySelector('mutation') ||
+    blockElement.ownerDocument.createElement('mutation');
+  // Place mutator before fields, values, and other nested blocks.
+  blockElement.insertBefore(mutationElement, blockElement.firstChild);
+
+  // We need to keep track of the expected number of inputs in order to create them all.
+  // If not, it needs a static behavior id in order to be translatable
+  // (e.g. shared behaviors).
+  // In CDO Blockly, the 'usercreated' flag was set on the block. Google Blockly
+  // expects this kind of extra state in a mutator.
+  const inputCount = blockElement.getAttribute('inputcount');
+  mutationElement.setAttribute('items', inputCount);
+}
+
+/**
+ * A helper function designed to process each individual block in an XML tree.
+ * @param {Element} block - The XML element for a single block.
+ */
+function processAllBlocks(block) {
```

Comment:
the function title is a little confusing, since it processes a single block (with potential child blocks), but is called `processAllBlocks`. Maybe `processBlockAndChildren` or something? Or just clarify in the comment we are processing a block and its children.

Reply:
It’s the block and it’s children. Thanks, I’ll clarify!